packages/drivers/odsp-driver/src/getFileLink.ts

```diff
@@ -153,16 +182,29 @@ async function getFileItemLite(
                 const requestInit = { method: "GET", headers };
                 const response = await fetchHelper(url, requestInit);
                 additionalProps = response.commonSpoHeaders;
-                if (response.content.ok) {
-                    return await response.content.json() as FileItemLite;
+
+                // We actually already know the response is ok, or else the fetchHelper would have thrown
+                // (it does an even more thorough check).  Keeping it here though for layers of protection.
+                if (!response.content.ok) {
+                    throwOdspNetworkError(
+                        `odspFetchError [${response.content.status}]`,
+                        response.content.status,
+                        response.content,
+                        await response.content.text(),
+                    );
                 }
-                return undefined;
+                const responseJson = await response.content.json();
+                if (!isFileItemLite(responseJson)) {
+                    // Here use a generic error.  fetchIncorrectResponse might be preferable but will result in a
+                    // retry, which we don't want.  Both because we don't have a reason to expect a different
+                    // response on retry, and also because the retry loop in in getFileLink doesn't have the same
+                    // "retry once" logic that getWithRetryForTokenRefresh does.
+                    throwOdspNetworkError("malformedGetFileItemLiteResponse", 400);
```

Comment:
I'm not sure. At a minimum, I'd add an error telemetry here, to ensure that we bring attention to this code if we hit it, giving a chance to engineers to re-evaluate this decision. Fundamentally it does not feel different from malformed JSON, in a sense that this is unexpected case that we do not well understand, but we know retrying there helps, and here we are about to fail fast, so retry once seems reasonable to start with

Reply:
Latest iteration will retry once.  Errors here will be logged in `getShareLinkPromise` under `FluidFileUrlError`.